Events/Controllers/EventController.cs

```diff
@@ -29,7 +29,130 @@ public EventController()
+        }
+        public async Task<Event> RetrieveEvent(String id)
+        {
+            
+            var request = new GetItemRequest
+            {
+                TableName = "Events",
+                Key = new Dictionary<string, AttributeValue>
+                 {
+                             { "id", new AttributeValue { S=id } }
+                 }
+            };
+            var response = await _dynamoDBClient.GetItemAsync(request);
+            Event Events = new Event
+            {
+                Id = response.Item["id"].S,
+                Name = response.Item["Name"].S,
+                Date = DateTime.Parse(response.Item["Date"].S),
+                Location = response.Item["Location"].S
+            };
+
+
+            return Events;
+        }
+        public async Task<string> AddEventAsync(Event _event)
+        {
+      
+            var request = new PutItemRequest
+            {
+                TableName = "Events",
+                Item = new Dictionary<string, AttributeValue>
+                 {
+                             { "id", new AttributeValue { S=_event.Id } },
+                             { "Name", new AttributeValue {S=_event.Name} },
+                             { "Location", new AttributeValue {S=_event.Location} },
+                             { "Date", new AttributeValue {S=_event.Date.ToString()} },
+
+                 }
+            };
+            await _dynamoDBClient.PutItemAsync(request);
+            return "Event Added Successfully";
+        }
+        public async Task<string> UpdateEvent( Event _event)
+        {
+
+        
+            if (_event == null)
+            {
+               
+                return "Event is null";
+            }
+
+            var ret = await RetrieveEvent(_event.Id);
+            if (ret != null)
+            {
+                var request1 = new UpdateItemRequest
+                {
+                    TableName = "Events",
+                    Key = new Dictionary<string, AttributeValue>
+                    {
+                             { "id", new AttributeValue { S=_event.Id } }
+                    },
+                    ExpressionAttributeNames = new Dictionary<string, string>
+                          {
+                            { "#attr1", "Location" },
+                            { "#attr2", "Date" }
+                          },
+                    ExpressionAttributeValues = new Dictionary<string, AttributeValue>
+                         {
+                            { ":val1", new AttributeValue { S = _event.Location } },
+                            { ":val2", new AttributeValue { S = _event.Date.ToString() } }
+                         },
+                    UpdateExpression = "SET #attr1 = :val1, #attr2 = :val2"
+                };
+
+                await _dynamoDBClient.UpdateItemAsync(request1);
+                return "Event Updated Successfully";
+            }
+           
+            return "Id is not found";

```

Comment:
Responding with not found should be a 404 HTTP response. This message will respond with 200 with this message.
Instead, you need to you `NotFound()`.